src/rt/dwarfeh.d

```diff
@@ -472,13 +472,25 @@ ClassInfo getClassInfo(_Unwind_Exception* exceptionObject)
 {
     ExceptionHeader* eh = ExceptionHeader.toExceptionHeader(exceptionObject);
     Throwable ehobject = eh.object;
+    auto currentLsd = eh.languageSpecificData;
     //printf("start: %p '%.*s'\n", ehobject, ehobject.classinfo.info.name.length, ehobject.classinfo.info.name.ptr);
     for (ExceptionHeader* ehn = eh.next; ehn; ehn = ehn.next)
     {
+        // like __dmd_personality_v0, don't combine when the exceptions are from different functions
+        // (fixes issue 19831, exception thrown and caught while inside finally block)
```

Comment:
Note, if a function has been partitioned, this will also not match.

However this will never happen with dmd, and gdc turns off partitioning unless the optimization is explicitly requested on the command line.

Reply:
....

[_throws up hands_ I GUESS--](https://i.imgur.com/Kp6RN00.png)

Reply:
This broke `runnable/eh.d` for LDC [with its slight modifications in this file]. The testsuite passes again when NOT breaking here as long as `currentLsd` is null, but I have no idea whether that's correct. @ibuclaw: any thoughts?

Reply:
Why would it be null?

Reply:
AFAICT, each `ExceptionHeader.languageSpecificData` starts out with null and is only set **after** finding a matching handler/catch in the search phase.

Reply:
Well the lsda is known since the beginning of the personality function, you could either eagerly set it, or pass it onto getClassInfo to initialize cuurentLsd with.

Reply:
Yeah, the question is rather 'why does DMD have no problem with it'. I'll pass it through; pls ping me if GDC has problems too, so that we can come up with something here upstream.